### DISCUSSION TITLE
zh/datasets/detect/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#24355] [Docs]
# zh/datasets/detect/

了解与 Ultralytics YOLO 兼容的数据集格式，以实现强大的目标检测。探索支持的数据集并学习如何转换格式。

https://docs.ultralytics.com/zh/datasets/detect/

**Comment by orz-serein** [comment]
Hello, I would like to ask a question. During the stage of preparing the training dataset, if some images do not have the target to be detected, do I need to add corresponding empty label files in the label folder?

**Comment by UltralyticsAssistant** [comment]
👋 Hello，感谢你对 Ultralytics 的关注 🚀！看起来这个讨论与 Ultralytics YOLO 检测数据集文档相关。对于新用户，我们建议先查看 [Docs](https://docs.ultralytics.com/)，其中包含大量 [Python](https://docs.ultralytics.com/usage/python/) 和 [CLI](https://docs.ultralytics.com/usage/cli/) 使用示例，很多常见问题都可以在文档中直接找到答案。你分享的中文页面 [zh/datasets/detect](https://docs.ultralytics.com/zh/datasets/detect/) 对了解 YOLO 兼容的数据集格式很有帮助 📚
这是一次自动回复 🤖，Ultralytics 工程师也会尽快前来协助。
如果这是一个 🐛 Bug Report，请提供一个 [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) 以便我们进行排查。
如果这是一个自定义训练 ❓ Question，请尽可能提供更多信息，包括数据集图像示例、训练日志，并确认你已参考我们的 [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)。
Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.
## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```
## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by raimbekovm** [comment]
hey @orz-serein, short answer: **no, you don't need empty label files for background images** - just leave the `.txt` out.

i verified this empirically on the current main branch. setup: 5 images, 2 with real labels, 1 with an empty `.txt` file, 2 without any `.txt` file at all. building a `YOLODataset` over them gives:

```
Scanning /tmp/bgtest_full/labels... 3 images, 3 backgrounds, 0 corrupt: 100% 5/5
```

all 5 images are kept. the 3 without labels (the empty-`.txt` one and the two with no `.txt` at all) all end up with the same `(0, 5)` labels array - they are treated identically as background images.

so both options work:

- **no `.txt` file**: counted as "missing" (`nm=1`), no labels -> background
- **empty `.txt` file**: counted as "empty" (`ne=1`), no labels -> background

both end up in the same `backgrounds = nm + ne` bucket in [`ultralytics/data/dataset.py`](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/data/dataset.py).

this is also stated in the [detection dataset docs](https://docs.ultralytics.com/datasets/detect/#ultralytics-yolo-format):

> If there are no objects in an image, no `*.txt` file is required.

**how many background images**: the [yolov5 training tips](https://docs.ultralytics.com/yolov5/tutorials/tips_for_best_training_results/) recommend 0-10% of the dataset as background images to reduce false positives (coco has about 1% background).

hope this helps!
